homeassistant/components/tplink/common.py

```diff
@@ -153,7 +153,5 @@ def add_available_devices(hass, device_type, device_class, async_add_entities):
                 ex,
             )
 
+    hass.data[TPLINK_DOMAIN][f"{device_type}_ready"] = entities_ready
```

Comment:
We can return the entities instead of storing them.

Reply:
Cool, changing to returning it.